helpers/integration_test.go

```diff
@@ -0,0 +1,192 @@
+					"build.appstudio.openshift.io/component": "component-sample",
+				},
+				Annotations: map[string]string{
+					"appstudio.redhat.com/updateComponentOnSuccess": "false",
+				},
+			},
+			Spec: tektonv1beta1.PipelineRunSpec{
+				PipelineRef: &tektonv1beta1.PipelineRef{
+					Name:   "build-pipeline-pass",
+					Bundle: "quay.io/kpavic/test-bundle:build-pipeline-pass",
+				},
+				Params: []tektonv1beta1.Param{
+					{
+						Name: "output-image",
+						Value: tektonv1beta1.ArrayOrString{
+							Type:      "string",
+							StringVal: "quay.io/redhat-appstudio/sample-image",
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, testpipelineRun)).Should(Succeed())
+
+		testpipelineRun.Status = tektonv1beta1.PipelineRunStatus{
+			PipelineRunStatusFields: tektonv1beta1.PipelineRunStatusFields{
+				TaskRuns: map[string]*tektonv1beta1.PipelineRunTaskRunStatus{
+					"index1": &tektonv1beta1.PipelineRunTaskRunStatus{
+						PipelineTaskName: "build-container",
+						Status: &tektonv1beta1.TaskRunStatus{
+							TaskRunStatusFields: tektonv1beta1.TaskRunStatusFields{
+								TaskRunResults: []tektonv1beta1.TaskRunResult{
+									{
+										Name:  "IMAGE_DIGEST",
+										Value: "image_digest_value",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Status().Update(ctx, testpipelineRun)).Should(Succeed())
+
+		Eventually(func() error {
+			err := k8sClient.Get(ctx, types.NamespacedName{
+				Name:      hasSnapshot.Name,
+				Namespace: "default",
+			}, hasSnapshot)
+			return err
+		}, time.Second*10).ShouldNot(HaveOccurred())
+	})
+	AfterEach(func() {
+		err := k8sClient.Delete(ctx, hasSnapshot)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+		err = k8sClient.Delete(ctx, testpipelineRun)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	AfterAll(func() {
+		err := k8sClient.Delete(ctx, hasApp)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	It("should return all IntegrationTestScenarios used by the application", func() {
+		updatedSnapshot, err := gitops.MarkSnapshotAsPassed(k8sClient, ctx, hasSnapshot, "test passed")
+		Expect(err == nil && updatedSnapshot != nil).To(BeTrue())
+		Expect(gitops.HaveHACBSTestsFinished(hasSnapshot)).To(BeTrue())
+
+		integrationTestScenarios, err := helpers.GetAllIntegrationTestScenariosForApplication(k8sClient, ctx, hasApp)
+		Expect(err == nil && integrationTestScenarios != nil).To(BeTrue())
+
+		for _, integrationTestScenario := range *integrationTestScenarios {
+			integrationTestScenario := integrationTestScenario //G601
+			integrationPipelineRun, err := helpers.GetLatestPipelineRunForApplicationSnapshotAndScenario(k8sClient, ctx, hasApp, hasSnapshot, &integrationTestScenario)
+			Expect(err != nil && integrationPipelineRun == nil)
+		}
+		pipelineRunOutcome, err := helpers.CalculateIntegrationPipelineRunOutcome(logger, testpipelineRun)
```

Comment:
After giving this some thought, I think we'll need to also have separate unit tests for the `CalculateIntegrationPipelineRunOutcome` function. 
Since it's such a key part of the integration service logic, I think it would be a good idea to have unit tests for three different pipelineRuns:

- A `pipelineRun` with one (or more) tasks which have the `HACBS_TEST_OUTPUT` Tekton result with a `SUCCESS` result
- A `pipelineRun` with at least one task which has the `HACBS_TEST_OUTPUT` Tekton result with a `FAILED` result
- A `pipelineRun` with 0 tasks that have the `HACBS_TEST_OUTPUT` result

Reply:
Added the above checks

Reply:
Thanks for adding the checks, but we'll need to modify the `testPipelineRun` to have `HACBS_TEST_OUTPUT` task results in order to test the decision logic.

Reply:
I added comments to the relevant places, resolving this part of the discussion